ext/NormalizingFlowsReverseDiffExt.jl

```diff
@@ -0,0 +1,22 @@
+module NormalizingFlowsReverseDiffExt
+
+if isdefined(Base, :get_extension)
+    using NormalizingFlows
+    using NormalizingFlows: ADTypes, DiffResults
+    using ReverseDiff
+else
+    using ..NormalizingFlows
+    using ..NormalizingFlows: ADTypes, DiffResults
+    using ..ReverseDiff
+end
+
+# ReverseDiff without compiled tape
```

Comment:
We could probably support compilation of the tape using some shenanigans with Memoize.jl or something (to avoid `compile` re-compiling the tape on every call to `value_and_gradient`). 

But let's not bother with this for now :+1: Maybe make an issue to keep track of it or smth?